test/integration/service/loadbalancer_test.go

```diff
@@ -138,3 +143,70 @@ func serviceHasNodePorts(svc *corev1.Service) bool {
 
 	return false
 }
+
+// Test_ServiceLoadBalancerEnableLoadBalancerClass tests that when a LoadBalancer
+// type of service has spec.LoadBalancerClass set, cloud provider should not create default load balancer.
+func Test_ServiceLoadBalancerEnableLoadBalancerClass(t *testing.T) {
```

Comment:
Can we also add a test to ensure loadBalancerClass is immutable after creation as long as the Service type is LoadBalancer?